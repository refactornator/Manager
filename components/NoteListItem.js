import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import timeago from 'timeago.js';

export default ({ note }) => (
  <View style={styles.item}>
    <Text style={styles.timestamp}>{timeago().format(note.createdAt)}</Text>
    <Text style={styles.text}>{note.text}</Text>
  </View>
);

const styles = StyleSheet.create({
  item: {
    flex: 1,
    padding: 2,
    marginTop: 2,
    borderColor: 'grey',
    borderWidth: 2
  },
  timestamp: {
    marginBottom: 2
  },
  text: {
    fontSize: 20
  }
});
