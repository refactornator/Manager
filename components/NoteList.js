import { FlatList } from 'react-native';
import React, { Component } from 'react';

import NoteListItem from './NoteListItem';

export default class NoteList extends Component {
  renderItem = ({ item: note }) => {
    const { editHandler, deleteHandler } = this.props;
    return <NoteListItem {...{ note, editHandler, deleteHandler }} />;
  };

  render() {
    const { notes } = this.props;
    return <FlatList data={notes} renderItem={this.renderItem} />;
  }
}
