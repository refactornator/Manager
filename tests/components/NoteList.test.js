import React from 'react';

import { shallow } from 'enzyme';

import NoteList from '../../components/NoteList';
import NoteListItem from '../../components/NoteListItem';

const notes = [{ key: 1, text: 'This is my note' }];
const props = {
  notes,
  editHandler: () => {},
  deleteHandler: () => {}
};

let render;
beforeEach(() => {
  render = shallow(<NoteList {...props} />);
});

it('renders a list of notes', () => {
  expect(render).toMatchSnapshot();
});

it('renders each note', () => {
  const item = notes[0];

  const received = JSON.stringify(render.instance().renderItem({ item }));
  const expected = JSON.stringify(
    <NoteListItem note={item} editHandler={() => {}} deleteHandler={() => {}} />
  );

  expect(received).toBe(expected);
});
