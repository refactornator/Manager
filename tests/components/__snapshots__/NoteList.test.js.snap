// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of notes 1`] = `
<FlatList
  data={
    Array [
      Object {
        "key": 1,
        "text": "This is my note",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
